lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -302,16 +312,22 @@ def write(chunk)
             entry['metadata']['severity'] = 'DEFAULT'
           end
 
+          # TODO(a-robinson): Consider falling back to parsing the namespace,
+          # pod, and container names out of the log name if we don't have this
+          # information, which depends on the kubernetes_metadata_filter
+          # filter plugin.
+          if @service_name == CONTAINER_SERVICE && record['kubernetes']
+            handle_container_metadata(record, entry)
+          end
+
           # If a field is present in the label_map, send its value as a label
           # (mapping the field name to label name as specified in the config)
           # and do not send that field as part of the payload.
-          unless label_map.nil?
+          if @label_map
             labels = {}
+            labels = entry['metadata']['labels'] if entry['metadata']['labels']
             @label_map.each do |field, label|
-              next unless record.key?(field)
-              # label values are required to be strings.
-              labels[label] = record[field].to_s
-              record.delete(field)
+              field_to_label(record, field, labels, label)
             end
             entry['metadata']['labels'] = labels unless labels.empty?
```

Comment:
Although, I'm starting to wonder if it would be cleaner to get rid of storing labels in a local variable (here and in handle_container_metadata) in favor of:

adding 'labels' => {} where we initialize the entry
adding something below that does delete entry['metadata']['labels'] if entry['metadata']['labels'].empty?

I'll leave it up to you.